core/src/main/java/org/elasticsearch/action/support/broadcast/BroadcastResponse.java

```diff
@@ -71,6 +74,17 @@ public int getFailedShards() {
     }
 
     /**
+     * The REST status that should be used for the response
+     */
+    public RestStatus getStatus() {
```

Comment:
Since this is effects all response that inherit from BroadcastResponse - can you check that all of them do the right thing and filter shard not available failures? might be good to add an assertion in the constructors, looking at all the exceptions this get.